ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/view/CircleImageViewTest.java

```diff
@@ -0,0 +1,379 @@
+        .invoke("setup");
+  }
+
+  @Test
+  public void borderOverlayTest() throws Exception {
+    // Arrange
+    boolean borderOverlay = circleImageView.isBorderOverlay();
+    borderOverlay = !borderOverlay;
+
+    // Act
+    circleImageView.setBorderOverlay(borderOverlay);
+
+    // Assert
+    assertThat(circleImageView.isBorderOverlay(), is(equalTo(borderOverlay)));
+    PowerMockito.verifyPrivate(circleImageView)
+        .invoke("setup");
+  }
+
+  @Test
+  public void borderOverlayWithoutChange() throws Exception {
+    // Arrange
+    boolean borderOverlay = circleImageView.isBorderOverlay();
+
+    // Act
+    circleImageView.setBorderOverlay(borderOverlay);
+
+    // Assert
+    assertThat(circleImageView.isBorderOverlay(), is(equalTo(borderOverlay)));
+    PowerMockito.verifyPrivate(circleImageView, Mockito.never())
+        .invoke("setup");
+  }
+
+  @Test
+  public void disableCircularTransformTest() throws Exception {
+    // Arrange
+    boolean disableCircularTransform = circleImageView.isDisableCircularTransformation();
+    disableCircularTransform = !disableCircularTransform;
+
+    // Act
+    circleImageView.setDisableCircularTransformation(disableCircularTransform);
+
+    // Assert
+    assertThat(circleImageView.isDisableCircularTransformation(), is(equalTo(disableCircularTransform)));
+    PowerMockito.verifyPrivate(circleImageView)
+        .invoke("initializeBitmap");
+  }
+
+  @Test
+  public void disableCircularTransformTestWithoutChange() throws Exception {
+    // Arrange
+    boolean disableCircularTransform = circleImageView.isDisableCircularTransformation();
+
+    // Act
+    circleImageView.setDisableCircularTransformation(disableCircularTransform);
+
+    // Assert
+    assertThat(circleImageView.isDisableCircularTransformation(), is(equalTo(disableCircularTransform)));
+    PowerMockito.verifyPrivate(circleImageView, Mockito.never())
+        .invoke("initializeBitmap");
+  }
+
+  @Test
+  public void checkConstructor() throws Exception {
+    // Arrange
+    CircleImageView mock = PowerMockito.mock(CircleImageView.class);
+    PowerMockito.whenNew(CircleImageView.class)
+        .withArguments(context)
+        .thenReturn(mock);
+    PowerMockito.whenNew(CircleImageView.class)
+        .withArguments(context, attributeSet)
+        .thenReturn(mock);
+    PowerMockito.whenNew(CircleImageView.class)
+        .withArguments(context, attributeSet, DEF_STYLE)
+        .thenReturn(mock);
+
+    // Act
+    new CircleImageView(context);
+    new CircleImageView(context, attributeSet);
+    new CircleImageView(context, attributeSet, DEF_STYLE);
+
+    // Assert
+    PowerMockito.verifyNew(CircleImageView.class)
+        .withArguments(context);
+    PowerMockito.verifyNew(CircleImageView.class)
+        .withArguments(context, attributeSet);
+    PowerMockito.verifyNew(CircleImageView.class)
+        .withArguments(context, attributeSet, DEF_STYLE);
```

Comment:
❗️ This snippet of code mixes Act and Assert phases. Since this is against AAA guideline and "Split Personality" test smell ([https://tinyurl.com/ybxbgp79](https://tinyurl.com/ybxbgp79)) and severely affects readability, it is recommended to follow the example and establish AAA order.  

**[More details](https://confluence.devfactory.com/x/iQCIEw)**
          
## The solution in this case

**Note** When there are dependencies between phases and mixing is required then this is a clear sign that the test should be split into multiple tests.  
```Java  
@Test
public void checkConstructorWithoutAttributesSet() throws Exception {
  // Arrange
  CircleImageView mock = PowerMockito.mock(CircleImageView.class);
  PowerMockito.whenNew(CircleImageView.class)
      .withArguments(context);

  // Act
  new CircleImageView(context);

  // Assert
  PowerMockito.verifyNew(CircleImageView.class)
      .withArguments(context);
}

@Test
public void checkConstructorWithAttributesSet() throws Exception {
  // Arrange
  PowerMockito.whenNew(CircleImageView.class)
      .withArguments(context, attributeSet)
      .thenReturn(mock);

  // Act
  new CircleImageView(context, attributeSet);

  // Assert
  PowerMockito.verifyNew(CircleImageView.class)
      .withArguments(context, attributeSet);
}  
@Test
public void checkConstructorWithAttributesSetAndDefStyle() throws Exception {
   ...
}
```
 

<a id="a5274cc0-4e70-11e8-8f04-8143e47b96ef" href="none" codereview-uuid-reserved-tag />